gapil/compiler/plugins/encoder/test/test.cpp

```diff
@@ -72,11 +70,11 @@ inner_class* create_inner_class_ref(arena* a, ref** p) {
 }
 
 context* create_context(arena* arena) {
-  return gapil_encoder_module.create_context(arena);
+  context* ctx = (context_t*)gapil_alloc(arena, sizeof(context_t), 8);
+  ctx->arena = arena;
+  return ctx;
 }
 
-void destroy_context(context* ctx) {
-  return gapil_encoder_module.destroy_context(ctx);
-}
+void destroy_context(context* ctx) { gapil_free(ctx->arena, ctx); }
```

Comment:
Might be good to add null checks?

Reply:
No, if `ctx` is null it would be a coding error and the test should fail and `gapil_free` already handles `null`.